README.md

```diff
@@ -166,40 +170,248 @@ Remove a key (and subsequent value) from this object. For example, to delete the
 "year" property from our car:
 
 ``` handlebars
-<button onclick={{car.delete "year"}}>
+<button onclick={{action car.delete "year"}}>
   Remove Year
 </button>
 {{!clicking will result in {make: 'Ford', model: 'Mustang'}}}
 ```
 
 #### `put(key, value)`
-#### `set(value)`
+
+Add property with a given name and value to the object. It will update the property with the given value if it already exists.
+
+```handlebars
+<button onclick={{action car.put "color" "blue"}}>
+  Add color
+</button>
+```
+
+#### `set(obj)`
+
+Replace current object microstate with a new object microstate from given hash.
+
+```handlebars
+<button onclick={{action car.set (hash make="Toyota" model="Supra" year="1982")}}>
+  Update car
+</button>
+```
 
 ### `List`
 
+List microstate represents an indexed array of values. 
+
+```hbs
+{{let numbers=(List (array 1 2 3))}}
+
+{{each numbers as |item|}}
+  {{item}}
+{{/each}}
+```
+
 #### `concat(list)`
+
+Merges given array with existing list.
```

Comment:
Yeah, `making a new list with all of the items from list added to the end of the current list` is a better description. I updated to use that description for all list transitions.